### PR TITLE
LG-4698: Allow choosing of axis for zoom selection 

### PR DESCRIPTION
**Pull request by tsck**
## ✍️ Proposed changes

It was an ask from consumers that they be able to optionally turn zoom selection on and off for axis. This change primarily adds that.

Note, this also updates the component's props. Previously we decided to just allow the handler to turn off and on the select-ability. Since now there is an option to enable it by axis, it made sense to just create a `zoomSelect` prop, that configures zoom selection. I like this approach better as a) the name `zoomSelect` clearer communicates that we're not actually zooming on data, and b) it removes the implied enablement of the previous design that none of us were crazy about.

🎟 _Jira ticket:_ [LG-4698](https://jira.mongodb.org/browse/LG-4698)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

## 🧪 How to test changes
Added to storybook so it can now be tested there.

### CODE REVIEW

File: charts/core/src/Chart/hooks/useChart.ts [hunk 1]
```diff
@@ -64,78 +146,39 @@ export function useChart({
     };
     window.addEventListener('resize', resizeHandler);
 
-    function onInitialRender() {
```
Comment (tsck) [inline charts/core/src/Chart/hooks/useChart.ts:67]: This was moved outside of the useEffect for readability

File: charts/core/src/Chart/hooks/useChart.ts [hunk 2]
```diff
@@ -64,78 +146,39 @@ export function useChart({
...
-         * this button and want it turned on by default. This is done by dispatching
-         * an action to enable the "dataZoomSelect" feature, as if it were clicked.
-         */
-        chartInstance.dispatchAction({
-          type: 'takeGlobalCursor',
-          key: 'dataZoomSelect',
-          dataZoomSelectActive: true,
-        });
-
-        chartInstance.on('dataZoom', (params: any) => {
```
Comment (tsck) [inline charts/core/src/Chart/hooks/useChart.ts:90]: This callback was moved into a function `handleDataZoom` outside of the `useEffect` for readability.

File: charts/core/src/Chart/hooks/useChart.ts
Comment (tsck) [inline charts/core/src/Chart/hooks/useChart.ts:1]: This whole file should probably be refactored as it's become a bit hard to follow. For time-sake I didn't do that here, but would love to do it before wrapping up the epic if time permits.

Comment (greggb) [reply charts/core/src/Chart/hooks/useChart.ts:1]: 👍 It would be nice, at some point, to have an easy to reference import of types without digging into the `dist` dir.

File: charts/core/src/Chart/hooks/useChart.types.ts [hunk 3]
```diff
@@ -1,12 +1,29 @@
 import { Theme } from '@leafygreen-ui/lib';
 
 export interface ZoomSelectionEvent {
-  xAxis: { startValue: number; endValue: number };
-  yAxis: { startValue: number; endValue: number };
+  xAxis?: { startValue: number; endValue: number };
+  yAxis?: { startValue: number; endValue: number };
+}
+
+export interface zoomSelect {
```
Comment (shaneeza) [inline charts/core/src/Chart/hooks/useChart.types.ts:8]:
```suggestion
export interface ZoomSelect {
```

File: charts/core/src/Chart/hooks/useChart.ts [hunk 4]
```diff
@@ -41,17 +41,99 @@ echarts.use([
...
+       * this button and want it turned on by default. This is done by dispatching
+       * an action to enable the "dataZoomSelect" feature, as if it were clicked.
+       */
+      chartInstance.dispatchAction({
+        type: 'takeGlobalCursor',
+        key: 'dataZoomSelect',
+        dataZoomSelectActive: true,
+      });
+
+      chartInstance.on('dataZoom', handleDataZoom);
```
Comment (shaneeza) [inline charts/core/src/Chart/hooks/useChart.ts:123]: If I understand correctly, even though `onInitialRender` is only called once on initial render, `handleDataZoom` will always be called when someone zooms?

File: charts/core/src/Chart/hooks/useChart.ts
Comment (tsck) [reply charts/core/src/Chart/hooks/useChart.ts:1]: Agreed! I've noticed this a few times where I've neglected to export types. I'm going to throw a ticket on the epic to make sure I remember.

File: charts/core/src/Chart/hooks/useChart.ts (hunk 4 above)
Comment (tsck) [reply charts/core/src/Chart/hooks/useChart.ts:123]: Correct. We remove `onInitialRender` as a render callback, but we never remove `handleDataZoom` as a dataZoom callback.

**Review by shaneeza — APPROVED** [review]

File: charts/core/README.md [hunk 5]
```diff
@@ -86,10 +86,10 @@ Chart container component.
...
+| `groupId` _(optional)_      | Charts with the same `groupId` will have their tooltips synced across charts. | `string`                               |                                  |
 | `onChartReady`              | Callback to be called when chart is finished rendering.                       | `() => void`                           |                                  |
-| `zoomSelect` _(optional)_   | Enable zoom select (click and drag area selection) for either axis.           | `{ xAxis?: boolean; yAxis?: boolean }` | `{ xAxis: false, yAxis: false }` |
 | `onZoomSelect` _(optional)_ | Callback to be called when a zoom selection is made if enabled.               | `(e: ZoomSelectionEvent) => void`      |                                  |
-| `groupId` _(optional)_      | Charts with the same `groupId` will have their tooltips synced across charts. | `string`                               |                                  |
+| `zoomSelect` _(optional)_   | Enable zoom select (click and drag area selection) for either axis.           | `{ xAxis?: boolean; yAxis?: boolean }` | `{ xAxis: false, yAxis: false }` |
```
Comment (shaneeza) [inline charts/core/README.md:92]: Should we be more explicit that`onZoomSelect` will do nothing if nothing is passed to this prop?

Comment (tsck) [reply charts/core/README.md:92]: That was my intention when I noted "if enabled" in the `onZoomSelect` description. Does that seem like not enough?

I see this like `Code`'s `copyable` and `onCopy`. Someone could add an `onCopy` handler, and if they never enable `copyable` it won't do anything https://github.com/mongodb/leafygreen-ui/blob/main/packages/code/README.md

Comment (shaneeza) [reply charts/core/README.md:92]: I think if I didn't have any context, I would not know right away what "if enabled" meant. Maybe something like:

> "Callback to be called when a zoom selection is made if `zoomSelect` is enabled"

Comment (tsck) [reply charts/core/README.md]: Updated! That does seem clearer. I think my lack of concern was that like in the `Code` docs, we don't mention anything about it. To me it seems inferable, but I might be too close to it so this is a good point.
